Message-Id: <d6609084b0b81abc74dc9db0281cdd0e074df5d4.1641539058.git.geliang.tang@suse.com>
From: Gopal Tiwari <gtiwari@redhat.com>
Date: Wed, 14 Apr 2021 14:16:45 +0530
Subject: [PATCH] nvme: fix NULL derefence in nvme_ctrl_fast_io_fail_tmo_show/store
Git-commit: d6609084b0b81abc74dc9db0281cdd0e074df5d4
References: git-fixes
Patch-mainline: v5.13-rc1

Adding entry for dev_attr_fast_io_fail_tmo to avoid the kernel crash
while reading and writing the fast_io_fail_tmo.

Fixes: 09fbed636382 (nvme: export fast_io_fail_tmo to sysfs)
Signed-off-by: Gopal Tiwari <gtiwari@redhat.com>
Reviewed-by: Keith Busch <kbusch@kernel.org>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 drivers/nvme/host/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/nvme/host/core.c b/drivers/nvme/host/core.c
index 288ac47ff5b4..40f08e6325ef 100644
--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -3379,6 +3379,8 @@ static umode_t nvme_dev_attrs_are_visible(struct kobject *kobj,
 		return 0;
 	if (a == &dev_attr_reconnect_delay.attr && !ctrl->opts)
 		return 0;
+	if (a == &dev_attr_fast_io_fail_tmo.attr && !ctrl->opts)
+		return 0;
 
 	return a->mode;
 }
-- 
2.31.1

